runtime/lua/vim/_meta/vvars.lua

```diff
@@ -1,23 +1,758 @@
 --- @meta _
-
--- TODO(lewis6991): generate this and `:help vim-variable`
+-- THIS FILE IS GENERATED
+-- DO NOT EDIT
+error('Cannot require a meta file')
 
 --- @class vim.v
+vim.v = ...
+
+--- The command line arguments Vim was invoked with.  This is a
+--- list of strings.  The first item is the Vim command.
+--- See `v:progpath` for the command with full path.
+--- @type string[]
+vim.v.argv = ...
+
+--- Argument for evaluating 'formatexpr' and used for the typed
+--- character when using <expr> in an abbreviation `:map-<expr>`.
+--- It is also used by the `InsertCharPre` and `InsertEnter` events.
+--- @type any
+vim.v.char = ...
+
+--- The name of the character encoding of a file to be converted.
+--- Only valid while evaluating the 'charconvert' option.
+--- @type string
+vim.v.charconvert_from = ...
+
+--- The name of the character encoding of a file after conversion.
+--- Only valid while evaluating the 'charconvert' option.
+--- @type string
+vim.v.charconvert_to = ...
+
+--- The extra arguments ("++p", "++enc=", "++ff=") given to a file
+--- read/write command.  This is set before an autocommand event
+--- for a file read/write command is triggered.  There is a
+--- leading space to make it possible to append this variable
+--- directly after the read/write command. Note: "+cmd" isn't
+--- included here, because it will be executed anyway.
+--- @type string[]
+vim.v.cmdarg = ...
+
+--- Set like v:cmdarg for a file read/write command.  When a "!"
+--- was used the value is 1, otherwise it is 0.  Note that this
+--- can only be used in autocommands.  For user commands `<bang>`
+--- can be used.
+--- @type any
+vim.v.cmdbang = ...
+
+--- The current locale setting for collation order of the runtime
+--- environment.  This allows Vim scripts to be aware of the
+--- current locale encoding.  Technical: it's the value of
+--- LC_COLLATE.  When not using a locale the value is "C".
+--- This variable can not be set directly, use the `:language`
+--- command.
+--- See `multi-lang`.
+--- @type string
+vim.v.collate = ...
+
+--- Dictionary containing the most recent `complete-items` after
+--- `CompleteDone`.  Empty if the completion failed, or after
+--- leaving and re-entering insert mode.
+--- Note: Plugins can modify the value to emulate the builtin
+--- `CompleteDone` event behavior.
+--- @type any
+vim.v.completed_item = ...
+
 --- The count given for the last Normal mode command.  Can be used
 --- to get the count before a mapping.  Read-only.  Example:
+---
 --- ```vim
---- :map _x :<C-U>echo "the count is " .. v:count<CR>
---- ```
+---   :map _x :<C-U>echo "the count is " .. v:count<CR>
+--- <
```

Comment:
This looks a bit confusing

Reply:
Because `norm_text()` should be called after removing leading spaces.